RestaurantsNearMe/Data/CloudKitServiceActor.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  CloudKitServiceActor.swift
+//  RestaurantsNearMe
+//
+//  Created by Colin Evans on 2023-12-12.
+//
+
+import Foundation
+import Combine
+
+actor CloudKitServiceActor {
```

Comment:
Essentially we make sure only one thread can use the isFetching at once, which when mutated is updated via combine.

(then any subscriber can observer on the main thread if needed)